packages/embedded/src/loader.ts

```diff
@@ -0,0 +1,451 @@
+      this._onError(error);
+      return { error };
+    }
+  }
+
+  /** @internal */
+  protected _onLoad(bundle: Bundle, _initial: boolean): void {
+    this.options.onLoad?.(bundle.metadata);
+  }
+
+  /** @internal */
+  protected _onError(error: LoadError): void {
+    this.options.onError?.(error);
+  }
+}
+
+const notModified = new Error("HTTP 304");
+
+/**
+ * Options for creating a new {@link AutoUpdatingLoader}.
+ *
+ * @public
+ */
+export interface AutoUpdateOptions extends Options {
+  /**
+   * Whether to activate updated embedded policy decision point bundles as soon as they are downloaded.
+   *
+   * @remarks
+   * If `false`, new bundles will be downloaded automatically but not used to evaluate policy decisions until you call {@link AutoUpdatingLoader.activate}.
+   * This might be useful if you want to activate updates only on page transitions to avoid layout shifts in your application.
+   *
+   * To detect whether an update is available to activate, provide an {@link Options.onLoad} callback or check the {@link AutoUpdatingLoader.pending} property.
+   *
+   * @defaultValue `true`
+   */
+  activateOnLoad?: boolean;
+
+  /**
+   * The delay (in milliseconds) between successive requests to check for new embedded policy decision point bundles.
+   *
+   * @defaultValue `60000` (1 minute)
+   */
+  interval?: number;
+}
+
+/**
+ * Loads an embedded policy decision point bundle from a given URL, and polls for updates.
+ *
+ * @public
+ */
+export class AutoUpdatingLoader extends Loader {
+  private readonly activateOnLoad: boolean;
+  private readonly interval: number;
+  private _pending: Bundle | undefined;
+  private etag: string | undefined;
+  private running = true;
+  private abortController?: AbortController;
+  private timeout?: NodeJS.Timeout;
+
+  /**
+   * Load an embedded policy decision point bundle from a given URL.
+   *
+   * @param url - URL from which to stream bundles.
+   * @param options - additional settings.
+   *
+   * @remarks
+   * Bundle download URLs are available in the "Embedded" section of the "Decision points" page of your Cerbos Hub workspace.
+   *
+   * The bundle will be loaded in the background when the loader is created.
+   * If initial loading fails, then the first request from the client using the loader will throw an error.
+   * To detect failure to load the bundle before making any requests, provide an {@link Options.onError} callback or await the {@link Loader.active} method.
+   *
+   * Failure to load updates after the initial load will not cause requests from the client to throw errors,
+   * but errors will be passed to the {@link Options.onError} callback.
+   */
+  public constructor(
+    private readonly url: string | URL,
+    {
+      activateOnLoad = true,
+      interval = 60000,
+      ...options
+    }: AutoUpdateOptions = {},
+  ) {
+    super(url, options);
+    this.activateOnLoad = activateOnLoad;
+    this.interval = interval;
```

Comment:
Should we cap the minimum interval value to prevent users from providing a really small value?

Reply:
Yep, seems reasonable. What do you think should be the minimum? Maybe 10 seconds?

Reply:
Yeah, 10 seconds seems reasonable. I can't think of an application that would require more frequent updates.